src/components/CatalogOverlay/CatalogScatterComponent/CatalogScatterComponent.tsx

```diff
@@ -176,9 +180,12 @@ export class CatalogScatterComponent extends React.Component<WidgetProps> {
 
     // Single source selected
     private onSingleSourceClick = (event: Readonly<Plotly.PlotMouseEvent>) => {
+        const appStore = AppStore.Instance;
         const dragmode: DragMode[] = ["select", "lasso"];
         const inDragmode = dragmode.includes(this.widgetStore.dragmode);
-        if (event && event.points && event.points.length > 0 && inDragmode) {
+        if (event && event.points && event.points.length > 0 && inDragmode && appStore.catalogProfiles.size > 0) {
```

Comment:
perhaps we should use the `?` operator here? [optional chaning](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining)